component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -72,21 +95,40 @@ public Article(XWikiDocument articleDoc, XWikiContext context) throws XWikiExcep
 
   public Article(Document articleDoc, XWikiContext context) throws XWikiException,
       EmptyArticleException {
-    this(articleDoc.getObjects("XWiki.ArticleClass"), articleDoc.getSpace(), context);
+    this(articleDoc.getObjects("XWiki.ArticleClass"), articleDoc.getDocumentReference(), context);
   }
 
   public Article(List<com.xpn.xwiki.api.Object> objList, String space, XWikiContext context)
       throws XWikiException, EmptyArticleException {
+    this(objList, getDocRefForObjList(objList, space), context);
+  }
+
+  public Article(List<com.xpn.xwiki.api.Object> objList, DocumentReference docRef,
+      XWikiContext context) throws XWikiException, EmptyArticleException {
     super(context);
-    for (Iterator<com.xpn.xwiki.api.Object> iterator = objList.iterator(); iterator.hasNext();) {
-      com.xpn.xwiki.api.Object artObj = iterator.next();
-      init(artObj, space);
+    articleDocRef = docRef;
+    for (Object artObj : objList) {
+      init(artObj, docRef.getLastSpaceReference().getName());
     }
     if (articleObjMap == null) {
       throw new EmptyArticleException();
     }
   }
 
+  private static DocumentReference getDocRefForObjList(List<com.xpn.xwiki.api.Object> objList,
+      String space) {
+    for (Object artObj : objList) {
+      if (artObj != null) {
+        String fn = artObj.getName();
+        if (!fn.startsWith(space + ".")) {
+          fn = fn.replaceAll("^.*?(\\..*)$", space + "$1");
```

Comment:
I was thinking about subscribed blogs and wasn't sure it's always the same ... however I suppose there could be problems if this happened. I'll add a warning log.